p2p/module.go

```diff
@@ -119,16 +110,13 @@ func (m *p2pModule) Start() error {
 	m.logger = logger.Global.CreateLoggerForModule(m.GetModuleName())
 	m.logger.Info().Msg("Starting network module")
 
-	addrbookProvider := getAddrBookProvider(m)
-	currentHeightProvider := getCurrentHeightProvider(m)
-
 	cfg := m.GetBus().GetRuntimeMgr().GetConfig()
 
 	// TODO: pass down logger
 	if cfg.P2P.UseRainTree {
-		m.network = raintree.NewRainTreeNetwork(m.address, m.GetBus(), addrbookProvider, currentHeightProvider)
+		m.network = raintree.NewRainTreeNetwork(m.address, m.GetBus(), m.addrBookProvider, m.currentHeightProvider)
 	} else {
-		m.network = stdnetwork.NewNetwork(m.GetBus(), addrbookProvider, currentHeightProvider)
+		m.network = stdnetwork.NewNetwork(m.GetBus(), m.addrBookProvider, m.currentHeightProvider)
```

Comment:
They come from there because we registered them but it's not something that's enforced.  For testability etc. I believe that the bus should stay a dependency of the providers